lib/enkf/enkf_main.cpp

```diff
@@ -599,12 +599,11 @@ static int enkf_main_serialize_dataset( const ensemble_config_type * ens_config
   matrix_type * A   = serialize_info->A;
   int ens_size      = matrix_get_columns( A );
   int current_row   = 0;
-  int ikw = 0;
-  hash_iter_type* data_iter = local_dataset_alloc_iter(dataset);
-  serialize_info->active_size.resize( local_dataset_get_size( dataset ));
-  serialize_info->row_offset.resize( local_dataset_get_size( dataset ));
-  while (!hash_iter_is_complete(data_iter)) {
-    const char             * key = hash_iter_get_next_key(data_iter);
+  stringlist_type * unscaled_keys = local_dataset_alloc_unscaled_keys(dataset);
```

Comment:
I think we should stop using the libecl vectors in as much of an extent as possible. 
use `std::vector<std::string> unscaled_keys` instead and propagate it downwords in form of iterators
like:
`for( auto & key : unscaled_keys) { }`

Reply:
I have used `std::vector<std::string>`as container - but the iteration is still based on an index; because that index is used for other purposes also in the loop body. If you prefer I can change to a range based for and manually increment the index.